torchao/quantization/quant_primitives.py

```diff
@@ -300,7 +300,7 @@ def _quantize_affine_no_dtype_cast(
     elif zero_point_domain is None:
         # This case handles quantization for float8 we expect no zero point and no zero point domain
         assert zero_point is None, "zero_point should be None when zero_point_domain is None"
-        quant = torch.clamp(input * scale.reciprocal(), quant_min, quant_max)
+        quant = torch.clamp(torch.round(input * (1.0 / scale)), quant_min, quant_max)
```

Comment:
@jerryzh168 to confirm if this is OK.  It was needed to match behavior of other quantizer.

Reply:
hmmm, it might be fine as long as all the tests passes I think

Reply:
I think the tests do not pass because it is slightly different quantization logic.  It looks more sensible to me to round before truncating, but I can also drop this change.

We can do a perplexity study when moving from the other quantizer to this one in torchchat.  But I have narrowed done this as being the only numerical difference between the two.

Reply:
I see, we don't want to break tests I think, but if this is better for torchchat we can create a new quant primitive op or add a new option here I feel